src/utils.rs

```diff
@@ -0,0 +1,43 @@
+use ipnetwork::IpNetwork;
+use reqwest::Client;
+use std::io::Result;
+use std::net::IpAddr;
+use worker::*;
+
+include!(concat!(env!("OUT_DIR"), "/prefixes.rs"));
+
+pub async fn is_cloudflare(domain: &str) -> Result<bool> {
+    if let Ok(ip) = domain.parse::<IpAddr>() {
+        console_debug!("{} is ip", domain);
+        Ok(is_ip_cloudflare(ip))
+    } else {
+        match Client::new()
```

Comment:
Have you considered using Cloudflare Workers KV for caching? Requesting DNS every time will introduce latency.